sdk/go/auto/git.go

```diff
@@ -81,40 +83,55 @@ func setupGitRepo(ctx context.Context, workDir string, repoArgs *GitRepo) (strin
 		}
 	}
 
+	// *Repository.Clone() will do appropriate fetching given a branch name. We must deal with
+	// different varieties, since people have been advised to use these as a workaround while only
+	// "refs/heads/<default>" worked.
+	//
+	// If a reference name is not supplied, then .Clone will fetch all refs (and all objects
+	// referenced by those), and checking out a commit later will work as expected.
+	if repoArgs.Branch != "" {
+		refName := plumbing.ReferenceName(repoArgs.Branch)
+		switch {
+		case refName.IsRemote(): // e.g., refs/remotes/origin/branch
+			shorter := refName.Short() // this gives "origin/branch"
+			parts := strings.SplitN(shorter, "/", 2)
+			if len(parts) == 2 {
+				refName = plumbing.NewBranchReferenceName(parts[1])
```

Comment:
Could also verify that the first part is "origin"; anything else is surely a mistake.